### PR TITLE
ready checks for all upload components

### LINKED ISSUE #621: Upload image container needs tcp readiness probe
/kind bug

**What happened**:

There's a difference between a pod's phase being "Running" and the pod being marked as "Ready".  A running pod may or may not be ready. The ready condition is determined by the readiness probe. 

When performing an image upload, we need virtctl to be able to wait until the pod is marked as "ready" by the service endpoint. Without this it's possible to attempt to upload to a uploadserver pod that hasn't quite initialized yet, which will fail.

In order for virtctl to work properly, the uploadserver pod needs a TCP Readiness probe. 

**What you expected to happen**:

I'd expect the uploadserver pod to be marked ready once a tcp readiness probe confirms that the pod's process is actually up and listening on 8443. 

**How to reproduce it (as minimally and precisely as possible)**:

This is a race condition. We just see random failures in CI related to uploading images. I believe it's because we attempt to upload to the uploadserver pod before the process in the pod is actually listening. The only way to know the service inside the pod is listening is to add a readiness check.

### PR DESCRIPTION
**Pull request by mhenriks**
**What this PR does / why we need it**:

We need better synchronization between all the upload components.  From the client perspective (virtctl), make sure that the uploadserver pod is really ready to handle requests.  Also fortify the uploadproxy for clients that aren't checking pod statuses by testing the connection to the upload server pod before attempting an http request.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #621 

**Special notes for your reviewer**:

**Release note**:

```release-note
Make sure the uploadserver pod is ready to handle requests with a ready check
```

### CODE REVIEW

**Comment by mhenriks** [comment]
ci test please

**Comment by mhenriks** [comment]
ci test please

File: cmd/cdi-controller/controller.go [hunk 1]
```diff
@@ -193,9 +201,25 @@ func main() {
...
+	f, err := os.Create(readyFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	return nil
+}
+
+func deleteReadyFile() {
+	os.Remove(readyFile)
```
Comment (awels) [inline cmd/cdi-controller/controller.go]: Do we want to check for an error on the delete, not sure what we would do with the error if it happens. But not being able to delete the file is probably a problem.

Comment (mhenriks) [reply cmd/cdi-controller/controller.go]: Useless info IMO.  Will just make code uglier and decrease coverage numbers.

Comment (awels) [reply cmd/cdi-controller/controller.go]: I missed this is basically the last thing before exit, can't even do anything useful with an error message if we got one.

**Comment by mhenriks** [comment]
ci test please

**Comment by mhenriks** [comment]
ci test please

**Comment by mhenriks** [comment]
ci test please

**Comment by mhenriks** [comment]
Investigating "make cluster-sync" timing out

**Pull request merged**